### ISSUE TITLE
[BUG] Contenttypes with slugs in snake_case aren't getting handled correctly

### ISSUE THREAD

**Issue by robbert-vdh** [bolt/bolt#5270]
Contenttypes with slugs containing underscores (e.g. `foo_bars`) won't hydrate correctly. If the contenttype doesn't have an explicitly set slug Bolt will hydrate the entity incorrectly and throw an error when trying to save (Bolt will somehow convert the snake_case to kebab-case).

````yaml
foo_bars:
#    slug: foo_bars
    name: foobars
    singular_slug: foo_bar
    singular_name: foobar
    fields:
        thing:
            type: text
        otherthing:
            type: integer
````
````
Content {#2054 ▼
  #contenttype: ContentType {#2038 ▼
    #boltname: "foo_bars"
    #contentType: array:14 [▼
      "name" => "foobars"
      "singular_slug" => "foo_bar"
      "singular_name" => "foobar"
      "fields" => array:2 [▼
        "thing" => array:6 [▶]
        "otherthing" => array:6 [▶]
      ]
      "slug" => "foo-bars"
      "show_on_dashboard" => true
      "show_in_menu" => true
      "sort" => false
      "default_status" => "draft"
      "viewless" => false
      "liveeditor" => false
      "tablename" => "foo_bars"
      "allow_numeric_slugs" => false
      "groups" => []
    ]
  }
  #_legacy: ContentLegacyService {#2069 ▶}
  #id: 1
  #slug: "thing"
  #datecreated: Carbon {#2016 ▶}
  #datechanged: Carbon {#2015 ▶}
  #datepublish: Carbon {#2014 ▶}
  #datedepublish: Carbon {#2013 ▶}
  #ownerid: null
  #status: ""
  #relation: null
  #taxonomy: null
  #group: null
  #sortorder: null
  +_fields: array:1 [▼
    "username" => ""
  ]
  #_internal: array:1 [▶]
  +"templatefields": []
}
````
````
Bolt\Exception\StorageException thrown with message "Attempted to load mapping data for unmapped class foo-bars"
````

If the contenttype does have an explicitly set slug, then `Bolt\Legacy\Storage::getContentType($slug)` will return false and break every query.

````
( ! ) Fatal error: Uncaught TypeError: Argument 2 passed to Bolt\Storage\Mapping\ContentType::__construct() must be of the type array, boolean given, called in /app/vendor/bolt/bolt/src/Storage/ContentLegacyService.php on line 51 and defined in /app/vendor/bolt/bolt/src/Storage/Mapping/ContentType.php on line 24
( ! ) TypeError: Argument 2 passed to Bolt\Storage\Mapping\ContentType::__construct() must be of the type array, boolean given, called in /app/vendor/bolt/bolt/src/Storage/ContentLegacyService.php on line 51 in /app/vendor/bolt/bolt/src/Storage/Mapping/ContentType.php on line 24
````

I was hoping this was the same problem as in # #5117, but I guess not (tested on 3.0-rc1).

**Comment by robbert-vdh** [comment]
Finally had a moment to look into this myself. The problem is that the content entities call the old `\Bolt\Storage\ContentLegacyService::setupContenttype` function, which uses `\Bolt\Legacy\Storage::getContentType` to retrieve information about the contenttype from the configuration.  `\Bolt\Legacy\Storage::getContentType` uses Slugify to create a slug version of the contenttype name, and Slugify replaces underscores with dashes by default. This could be fixed by making Slugify use underscores when dealing with contenttypes and taxonomies, but that would probably break existing functionality.

The two more sensible options are either including an additional check for the non-slugified version of the content type, or slugifying _everything_ before comparing, like a case-insensitive comparison. I've created a patch for the former and I'll submit a pull request after I've added some tests.

**Comment by GawainLynch** [comment]
Yeah, this shouldn't be too hard to fix… and legacy is about to face another round of "Ross" soon.

A lot of the delay in getting to things like this was tonight's release of 3.0.0 (and now the core team is going to take a few days of well earned rest) :smile_cat:

**Comment by robbert-vdh** [comment]
Yeah, congratulations on the release! I'll just run my own fork until this gets merged in Bolt 3.0.X. This is probably the last quirk in Bolt 3.0 I've noticed.

**Comment by GawainLynch** [comment]
Zero stress… I was a bit bummed myself, I have a couple of branches in motion that would have taken use to Slugify 2.x and fixed a few of these things… but… real life and time constraints… and :beers:
